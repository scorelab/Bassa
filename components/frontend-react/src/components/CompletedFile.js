import React from 'react';
import moment from 'moment';

// MUI imports
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Button from '@material-ui/core/Button';
import SaveIcon from '@material-ui/icons/Save';

const CustomTableCell = withStyles(theme => ({
  head: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white
  },
  body: {
    fontSize: 14
  }
}))(TableCell);

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
    overflowX: 'auto'
  },
  row: {
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.background.default
    }
  },
  button: {
    margin: theme.spacing(2)
  },
  icon: {
    marginRight: theme.spacing(2),
    fontSize: 20
  }
}));

const CompletedFile = props => {
  const classes = useStyles();
  const { file, onDownload } = props;
  const timeStamp = new Date(file.completed_time * 1000);
  return (
    <div className={classes.root}>
      <Table>
        <TableBody>
          <TableRow>
            <CustomTableCell>{file.id}</CustomTableCell>
            <CustomTableCell className="row" component="th" scope="row">
              {file.user_name}
            </CustomTableCell>
            <CustomTableCell align="left">{file.download_name}</CustomTableCell>
            <CustomTableCell align="left">
              {file.size / 1024} KB
            </CustomTableCell>
            <CustomTableCell align="left">
              {moment(timeStamp).fromNow()}
            </CustomTableCell>
            <CustomTableCell align="right">
              <Button
                data-test="button-download"
                variant="outlined"
                size="small"
                color="primary"
                className={classes.button}
                onClick={onDownload}
              >
                <SaveIcon className="icon" /> Download
              </Button>
            </CustomTableCell>
          </TableRow>
        </TableBody>
      </Table>
    </div>
  );
};

export default CompletedFile;
