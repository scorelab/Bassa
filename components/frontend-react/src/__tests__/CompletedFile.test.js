/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import CompletedFile from '../components/CompletedFile';

import fileObject from '../stories/CompletedFile.stories';

const clickDownloadFn = jest.fn();

const findByTestAttr = (component, attr) => {
  const wrappr = component.find(`[data-test='${attr}']`);
  return wrappr;
};

describe('Completed single file component', () => {
  let component;
  beforeEach(
    // eslint-disable-next-line no-return-assign
    () =>
      (component = shallow(
        <CompletedFile file={fileObject} onDownload={clickDownloadFn()} />
      ))
  );

  it('should match with its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call onDownload function on click', () => {
    const wrapper = findByTestAttr(component, 'button-download');
    wrapper.simulate('click');
    expect(clickDownloadFn).toHaveBeenCalled();
  });
});
