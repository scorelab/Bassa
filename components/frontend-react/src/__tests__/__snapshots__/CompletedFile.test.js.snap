// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Completed single file component should match with its snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(WithStyles(ForwardRef(TableCell)))>
          1
        </WithStyles(WithStyles(ForwardRef(TableCell)))>
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          className="row"
          component="th"
          scope="row"
        />
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          align="left"
        />
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          align="left"
        >
          NaN
           KB
        </WithStyles(WithStyles(ForwardRef(TableCell)))>
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          align="left"
        />
        <WithStyles(WithStyles(ForwardRef(TableCell)))
          align="right"
        >
          <WithStyles(ForwardRef(Button))
            className="makeStyles-button-3"
            color="primary"
            data-test="button-download"
            size="small"
            variant="outlined"
          >
            <SaveIcon
              className="icon"
            />
             Download
          </WithStyles(ForwardRef(Button))>
        </WithStyles(WithStyles(ForwardRef(TableCell)))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
</div>
`;
